src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -1385,7 +1385,13 @@ private int ReadBytes(Encoding encoding, int byteBlock, int charBlock, byte[] bu
                 {
                     int actualCharCount;
                     if (readContent)
+                    {
                         actualCharCount = ReadContentAsChars(chars, charCount, maxCharCount - charCount);
+                        // When deserializing base64 content which contains new line chars (CR, LF) chars from ReadObject, the reader reads in chunks of base64 content, LF char, base64 content, LF char and so on
+                        // Relying on encoding.GetBytes' exception to handle LF char would result in performance degradation so skipping LF char here
+                        if (actualCharCount == 1 && chars[charCount] == '\n')
+                            continue;
+                    }
                     else
                         actualCharCount = ReadValueChunk(chars, charCount, maxCharCount - charCount);
```

Comment:
:bulb: Consider adding braces around the `else` clause since you added them to the `if` clause.